docs/architecture/workflow-update.md

```diff
@@ -155,14 +146,61 @@ Update Registry intact. But it proved to be too error-prone.
+| **ProvisionallyAdmitted**               | `WorkflowUpdateAbortedErr` | `ErrWorkflowCompleted`                   | `ErrWorkflowClosing`                     |
+| **Admitted**                            | `WorkflowUpdateAbortedErr` | `ErrWorkflowCompleted`                   | `ErrWorkflowClosing`                     |
+| **Sent**                                | `WorkflowUpdateAbortedErr` | `ErrWorkflowCompleted`                   | `ErrWorkflowClosing`                     |
+| **ProvisionallyAccepted**               | `WorkflowUpdateAbortedErr` | `acceptedUpdateCompletedWorkflowFailure` | `acceptedUpdateCompletedWorkflowFailure` |
+| **Accepted**                            | `WorkflowUpdateAbortedErr` | `acceptedUpdateCompletedWorkflowFailure` | `acceptedUpdateCompletedWorkflowFailure` |
+| **ProvisionallyCompleted**              | `WorkflowUpdateAbortedErr` | `acceptedUpdateCompletedWorkflowFailure` | `acceptedUpdateCompletedWorkflowFailure` |
+| **ProvisionallyCompletedAfterAccepted** | `WorkflowUpdateAbortedErr` | `acceptedUpdateCompletedWorkflowFailure` | `acceptedUpdateCompletedWorkflowFailure` |
+| **Completed**                           | `nil`                      | `nil`                                    | `nil`                                    |
+| **Aborted**                             | `nil`                      | `nil`                                    | `nil`                                    |
+
+When the Workflow performs a final completion, all in-flight Updates are aborted: admitted Updates get
+`ErrWorkflowCompleted` error on both `accepted` and `completed` futures. Accepted Updates
+are failed with special server `acceptedUpdateCompletedWorkflowFailure` failure because if a client
+knows that Update has been accepted, it expects any following requests to return an Update result
+(or failure) but not an error. This failure is set on the `completed` future only. 
+
+When a Workflow completion command creates a new run, accepted Updates are failed in the same way:
+with the `acceptedUpdateCompletedWorkflowFailure` failure on the `completed` future. Admitted Updates,
+though, are aborted with the retryable `ErrWorkflowClosing` error. Server internally retries this error
+and new attempt should land on the new run. Because Updates received while the Workflow Task
+was running haven't been seen by the Workflow yet can be safely retried on the new run.
+It also provides a better experience for API callers since they will not notice that the Workflow
+started a new run.
+
+`WorkflowUpdateAbortedErr` is also retried internally by the server providing a better experience
+to the API caller: they will not notice that the Update was lost.
+
+`Aborted` is a terminal state. Updates remain in the `Aborted` state in the Registry even after
+the Update Registry is reconstructed from the history.
+
```

Comment:
Existing text:

> These can be thought of as "PUT + GET" and "GET", respectively.

Would `"get-or-create" or "get"` be clearer / more accurate?